src/main/java/org/reactome/addlinks/kegg/KEGGReferenceDatabaseGenerator.java

```diff
@@ -110,12 +116,12 @@ public static void generateSpeciesSpecificReferenceDatabases(ReferenceObjectCach
 		}
 		// Special cases for viruses and "addendum"
 		String speciesURL = KEGG_URL.replace("###SP3###", "vg:");
-		String newDBName = "KEGG Gene (Viruses)";
+		String newDBName = KEGG_GENE + " (Viruses)";
 		try
 		{
 			createReferenceDatabase(newDBName, "Viruses", speciesURL, objectCache);
 			speciesURL = KEGG_URL.replace("###SP3###", "ag:");
-			newDBName = "KEGG Gene (Addendum)";
+			newDBName = KEGG_GENE + " (Addendum)";
 			createReferenceDatabase(newDBName, "Addendum", speciesURL, objectCache);
 		}
```

Comment:
The `replace` function does not modify the string it is called on - it _returns_ the result of the replacement operation. So `KEGG_URL` does not get modified by calling replace (Strings, in Java, are immutable - https://docs.oracle.com/javase/tutorial/java/data/strings.html).

`newDBName` and `speciesURL` get new values only _after_ the first call to `createReferenceDatabase` (line 122) - at that point, the name and URL for "viruses" should be in the database so it's OK to reuse the variables with new values, and then put _those_ values into the database (line 125).

Reply:
Yup, you're right.

Okay, so since two reference databases are created with two distinct names and URL values, should the link checking on line 132 happen twice as well?  Right now it looks like it's only called for the "Addendum" db name and not "Viruses" db name since `newDBName` gets reassigned on line 124 before the link checking happens.

Reply:
Good catch. That's been updated.